### PR TITLE
adding more detailed logging

### PR DESCRIPTION
**Pull request by KDreynolds**
#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by tb582** [comment]
@ncw can we get this approved ? looking to test the build!

**Review by ncw — COMMENTED** [review]
I think this will log that a file is excluded twice for some operations (eg `rclone lsl --exclude...`)

Looking at https://github.com/rclone/rclone/issues/7463#issuecomment-1845780139

I think the logs in `fs/walk` need to be removed.

Can you confirm what happens with `rclone sync/copy/move/ls` and check that there is one and only one log made for each excluded file - thanks!

**Comment by KDreynolds** [comment]
Oh my mistake! I'll look at this here shortly.

**Comment by KDreynolds** [comment]
Ok, should be addressed now, thank you!

**Comment by tb582** [comment]
anything here @ncw

File: fs/filter/filter.go [hunk 1]
```diff
@@ -330,23 +330,30 @@ func (f *Filter) DirContainsExcludeFile(ctx context.Context, fremote fs.Fs, remo
...
 
 // Include returns whether this object should be included into the
-// sync or not
+// sync or not and logs the reason for exclusion if not included
 func (f *Filter) Include(remote string, size int64, modTime time.Time, metadata fs.Metadata) bool {
 	// filesFrom takes precedence
 	if f.files != nil {
 		_, include := f.files[remote]
+		if !include {
+			fs.Debugf(nil, "Excluded (FilesFrom Filter): %s", remote)
```
Comment (ncw) [inline fs/filter/filter.go:339]: These debugs should be

    fs.Debugf(remote, "Excluded (FilesFrom Filter)", remote)

Then they will appear correct in the logs.

**Comment by KDreynolds** [comment]
Hey @ncw Let me know if this is how you meant for me to update this. Thanks!

**Comment by tb582** [comment]
@ncw ? approval here

File: fs/filter/filter.go [hunk 2]
```diff
@@ -330,23 +330,30 @@ func (f *Filter) DirContainsExcludeFile(ctx context.Context, fremote fs.Fs, remo
...
 
 // Include returns whether this object should be included into the
-// sync or not
+// sync or not and logs the reason for exclusion if not included
 func (f *Filter) Include(remote string, size int64, modTime time.Time, metadata fs.Metadata) bool {
 	// filesFrom takes precedence
 	if f.files != nil {
 		_, include := f.files[remote]
+		if !include {
+			fs.Debugf(remote, "Excluded (FilesFrom Filter)", remote)
```
Comment (ncw) [inline fs/filter/filter.go:339]: These have got a spurious ", remote" on the end

**Review by ncw — COMMENTED** [review]
This is almost perfect but there are spurious `, remote` on the end of the debug statements which I think the CI will complain about. I've run it so we can see :-)

File: fs/filter/filter.go (hunk 2 above)
Comment (tb582) [reply fs/filter/filter.go:339]: changing it to` fs.Debugf("Excluded '%s' (FilesFrom Filter)", remote)` should work...

**Comment by KDreynolds** [comment]
ok I think I have it now. sorry.

File: fs/filter/filter.go (hunk 2 above)
Comment (ncw) [reply fs/filter/filter.go]: It should be

    fs.Debugf(remote, "Excluded (FilesFrom Filter)")

**Review by ncw — APPROVED** [review]
This is looking perfect now - thank you :-)

I'll just make sure the CI runs properly and then I'll merge!

**Comment by KDreynolds** [comment]
Thank you @ncw !

**Comment by tb582** [comment]
@ncw when does this get built into a beta build?

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.66

Thank you @KDreynolds for a nice improvement :-)

**Comment by KDreynolds** [comment]
Thank you both for bearing with me on this!

**Pull request merged**